packages/@aws-cdk/aws-appmesh/lib/gateway-route-spec.ts

```diff
@@ -150,21 +267,57 @@ class HttpGatewayRouteSpec extends GatewayRouteSpec {
     this.match = options.match;
   }
 
-  public bind(_scope: Construct): GatewayRouteSpecConfig {
-    const prefixPath = this.match ? this.match.prefixPath : '/';
-    if (prefixPath[0] != '/') {
-      throw new Error(`Prefix Path must start with \'/\', got: ${prefixPath}`);
+  public bind(scope: Construct): GatewayRouteSpecConfig {
+    const matchConfig = this.match?.path?.bind(scope);
+    const defaultTargetHostname = this.match?.rewriteRequestHostname;
+
+    // Set prefix Match to '/' if none on match properties are defined.
+    const prefixPathMatch = areMatchPropertiesUndefined(this.match)
+      ? '/'
+      : matchConfig?.prefixPathMatch;
```

Comment:
Same question here - the documented behavior and what the code actually does are different, for example for the following route:

```ts
gateway.addGatewayRoute('Route', {
  routeSpec: appmesh.GatewayRouteSpec.http({
    match: {
      method: appmesh.HttpRouteMethod.GET,
    },
  }),
});
```

Reply:
Please find [this comment](https://github.com/aws/aws-cdk/pull/15395#discussion_r665801551).

Reply:
I will update the validation and default to actually assign `prefix` with `/`.